### PR TITLE
[EAG2-22] Added findAvailableGamesTest

### PR DESCRIPTION
**Pull request by TarasDzivik**
I still do not understand how it should work, because the test successfully passed but I can't fail it.
I think that I made something wrong. And I need some advice before I will start to make another piece of the subtask.

### CODE REVIEW

File: src/main/java/com/eleks/academy/whoami/core/impl/PersistentGame.java [hunk 1]
```diff
@@ -50,6 +67,11 @@ public String getId() {
 	@Override
 	public SynchronousPlayer enrollToGame(String player) {
 		// TODO: Add player to players list
+		if (players.size() == maxPlayers){
```
Comment (VolodymyrKr-Eleks) [inline src/main/java/com/eleks/academy/whoami/core/impl/PersistentGame.java:70]: is this part of this task?

File: src/main/java/com/eleks/academy/whoami/core/impl/PersistentGame.java [hunk 2]
```diff
@@ -110,6 +128,20 @@ public void initGame() {
 
 	}
 
+	@Override
+	public boolean makeTurn() {
+		this.turnLock.lock();
```
Comment (VolodymyrKr-Eleks) [inline src/main/java/com/eleks/academy/whoami/core/impl/PersistentGame.java:133]: is this part of the current task?

**Review by VolodymyrKr-Eleks — CHANGES_REQUESTED** [review]

File: src/test/java/com/eleks/academy/whoami/service/GameServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        var gameLight = gameService.findAvailableGames(player);
+        assertThat(gameService.findAvailableGames(player)).usingRecursiveComparison()
+                .isEqualTo(gameLight);
+        assertThat(gameRepository.findAllAvailable(null)).isEqualTo(gameLight);
+    }
+
+
+
+    @Test
+    void createGameTest() {
```
Comment (VolodymyrKr-Eleks) [inline src/test/java/com/eleks/academy/whoami/service/GameServiceImplTest.java:46]: please do not create empty test/code that is not implemented

File: src/test/java/com/eleks/academy/whoami/service/GameServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+        gameRequest.setMaxPlayers(4);
+        gameRepository.findAllAvailable("Name");
+    }
+
+    @Test
+    void findAvailableGamesTest() {
+        final String player = "name";
+        var gameLight = gameService.findAvailableGames(player);
+        assertThat(gameService.findAvailableGames(player)).usingRecursiveComparison()
```
Comment (VolodymyrKr-Eleks) [inline src/test/java/com/eleks/academy/whoami/service/GameServiceImplTest.java:38]: what is the purpose of triggering the same method twice and checking only once? maybe let's trigger once and check once?

File: src/test/java/com/eleks/academy/whoami/service/GameServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                .isEqualTo(gameLight);
+        assertThat(gameRepository.findAllAvailable(null)).isEqualTo(gameLight);
```
Comment (VolodymyrKr-Eleks) [inline src/test/java/com/eleks/academy/whoami/service/GameServiceImplTest.java:40]: why do we need to test the repository in service test?

**Review by VolodymyrKr-Eleks — CHANGES_REQUESTED** [review]

File: src/test/java/com/eleks/academy/whoami/service/GameServiceImplTest.java (hunk 3 above)
Comment (TarasDzivik) [reply src/test/java/com/eleks/academy/whoami/service/GameServiceImplTest.java:46]: I just forgot to delete these autogenerated methods.

**Pull request closed without merging**